kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -270,6 +277,117 @@ def set_phase_parameters(
+        detector : EBSDDetector
+            EBSDDetector object describing the detector geometry.
+        energy : int
+            The wanted energy in the master pattern.
+        chunk_size : int
+            The amount of rotations the dask arrays should work on per chunk.
+        dtype_out : numpy.dtype
+            Data type of the returned patterns.
+
+        Returns
+        ----------
+        LazyEBSD object containing the simulated EBSD patterns.
+        """
+
+        if (
+            self.metadata.Simulation.EBSD_master_pattern.Master_pattern.projection
+            != "lambert"
+        ):
+            raise NotImplementedError(
+                "Method only supports master patterns in lambert projection!"
+            )
+
+        dc = _get_direction_cosines(detector)
+
+        n = rotations.size
+        det_y, det_x = detector.shape
+
+        out_shape = (n, det_y, det_x)
+        chunks = (min(chunk_size, n), det_y, det_x)
+
+        dtype_out = dtype_out
+
+        rescale = False
+        if dtype_out != np.float32:
+            rescale = True
+
+        r_da = da.from_array(rotations.data, chunks=(chunks[0], -1))
+
+        # 4 cases
+        # Has energies, has hemis - Case 1
+        if len(self.axes_manager.shape) == 4:
+            energies = self.axes_manager["energy"].axis
+            energy_index = (np.abs(energies - energy)).argmin()
+            mpn = self.data[0, energy_index]
+            mps = self.data[1, energy_index]
+
+        # no energies, no hemis - Case 2
+        elif len(self.axes_manager.shape) == 2:
+            mpn = self.data
+            mps = mpn
+        else:
+            try:  # has energies, no hemi - Case 3
+                energies = self.axes_manager["energy"].axis
+                energy_index = (np.abs(energies - energy)).argmin()
+                mpn = self.data[energy_index]
+                mps = mpn
+            except ValueError:  # no energies, yes hemi - Case 4
+                mpn = self.data[0]
+                mps = self.data[1]
+
+        npx, npy = self.axes_manager.signal_shape
+
+        simulated = r_da.map_blocks(
+            _get_patterns_chunk,
+            dc=dc,
+            master_north=mpn,
+            master_south=mps,
+            npx=npx,
+            npy=npy,
+            rescale=rescale,
+            dtype_out=dtype_out,
+            drop_axis=1,
+            new_axis=(1, 2),
+            chunks=chunks,
+            dtype=dtype_out,
+        )
+
+        names = ["x", "dy", "dx"]
+        scales = np.ones(3)
+
+        # Create axis objects for each axis
+        axes = [
+            {
+                "size": out_shape[i],
+                "index_in_array": i,
+                "name": names[i],
+                "scale": scales[i],
+                "offset": 0.0,
+                "units": "px",
+            }
+            for i in range(simulated.ndim)
+        ]
+
+        return LazyEBSD(simulated, axes=axes)
```

Comment:
Should we add a `compute: bool = False` parameter to this method, so that the user has the option to return an `EBSD` object directly from this computation?

Reply:
I like the idea of that!